### PR TITLE
Add Plugins Section

### PR DESCRIPTION
**Pull request by fpliger**

### CODE REVIEW

File: docs/user-guide/plugins.md [hunk 1]
```diff
@@ -1,3 +1,184 @@
...
+* main thread: executing code in the browser main thread means the code is being executed in the same
+process where the browser processes user events and paints. By default, the browser uses this single thread
+to run all the JavaScript code in your page, as well as to perform layout, reflows, and garbage collection.
+This means that long-running code or blocking calls can or will block the thread, leading to an unresponsive
+page and a bad user experience.
+* web workers: code executed in workers actually run on "background" threads. This means the code can perform
+tasks without interfering with the user interface or other operations being perfomed in the main thread. While
+this adds great flexibility it's important to understand that workers actually have limited capabilities when
+comparing to code executed on the main thread. For instace, while PyScript offers a DOM API that actually can
+be used in web workers the browser, by default, does not allow DOM operation in workers.
```
Comment (WebReflection) [inline docs/user-guide/plugins.md:84]: this is misleading ... we offer both `window` and `document` in the `pyscript` module that's available in either the *main* thread or the *worker* one. The non working with *DOM* namespace is the pyodide or MicroPython *js* one, which doesn't have a document at all in workers but the whole point of our `pyscript` module is to make this difference almost non existent.

File: docs/user-guide/plugins.md [hunk 2]
```diff
@@ -1,3 +1,184 @@
...
+At the moment, PyScript supports plugins written in Javascript. These plugins can use PyScript
+Plugins API to define entry points and hooks so that the plugin can be collected and hook into
+the PyScript lifecycle events, with the ablity to modify and integrate the features of PyScript
+core itself.
+
+Here's an example of how a PyScript plugin looks like:
+
+```js
+// import the hooks from PyScript first...
+import { hooks } from "<path_to_pyscript>/core.js";
```
Comment (WebReflection) [inline docs/user-guide/plugins.md:17]: A user already asked where to grab this `<path_to_pyscript>` so I suggest we use a real link here and update such link on every release.

**Review by WebReflection — CHANGES_REQUESTED** [review]
Please see my 2 comments, thanks!

File: docs/user-guide/plugins.md (hunk 1 above)
Comment (fpliger) [reply docs/user-guide/plugins.md:84]: Good point. I'll try to rephrase it better

File: docs/user-guide/plugins.md [hunk 3]
```diff
@@ -1,3 +1,187 @@
...
+This means that long-running code or blocking calls can or will block the thread, leading to an unresponsive
+page and a bad user experience.
+* web workers: code executed in workers actually run on "background" threads. This means the code can perform
+tasks without interfering with the user interface or other operations being perfomed in the main thread. While
+this adds great flexibility it's important to understand that workers actually have limited capabilities when
+comparing to code executed on the main thread. For instace, while PyScript offers a DOM API that actually can
+be used in web workers on the browser, by default, does not allow DOM operation in workers. So, in this case,
+if you just use `window` and `document` directly mapping the Javascript FFI provided directly by the interpreters
+we support (Pyodide and MicroPython): `from pyscript import window, document` will work and allow you to interact
+with the DOM while `from js import window, document` will not.
```
Comment (WebReflection) [inline docs/user-guide/plugins.md:87]: while `from js import document`, or `import js; js.document`, will not.

**Review by WebReflection — APPROVED** [review]
Minor suggestion as usually `from js import window` is not common or a good example while `document` is the one missing ... although `window`, strictly speaking, is not in workers neither so it might be OK as it is, thanks.

**Comment by fpliger** [comment]
I think I've addressed your comment and, if not, we can put a follow up PR...

**Pull request merged**